mmv1/products/compute/api.yaml

```diff
@@ -8902,6 +8902,35 @@ objects:
               Maximum size of the node group. Set to a value less than or equal
               to 100 and greater than or equal to min-nodes.
             required: true
+      - !ruby/object:Api::Type::NestedObject
+        name: 'shareSettings'
+        description: |
+          Share settings for the node group.
+        properties:
+        - !ruby/object:Api::Type::Enum
+          name: 'shareType'
+          required: true
+          description: |
+            Node group sharing type.
+          values:
+          - :ORGANIZATION
+          - :SPECIFIC_PROJECTS
+          - :LOCAL
+        - !ruby/object:Api::Type::Map
+          name: 'projectMap'
+          description: |
+            A map of project id and project config. This is only valid when shareType's value is SPECIFIC_PROJECTS.
+          key_name: id
+          key_description: |
+            The project ID.
+          value_type: !ruby/object:Api::Type::NestedObject
+            name: projectConfig
+            properties:
+            - !ruby/object:Api::Type::String
+              name: 'projectId'
+              required: true
+              description: |
+                The project id/number should be the same as the key of this project config in the project map.
```

Comment:
This seems to be implemented correctly in terms of the API, but I cannot understand the design. The `project_id` the only relevant information being passed here, why is it duplicated in both the key and value within a map? Am I missing something? Do you know if the API is setting up to have other values within the `projectConfig` nested object?

We rarely deviate from the API, but it seems like a frivolous user experience to have to set the project id twice. We could relatively simply implement custom code to take a list of project id's from the user and supply the API with the map in the correct format. From my perspective, this could prevent user error. More context would help me understand however

Reply:
To be honest, I cannot defend this decision, but I don't think it's been designed while working on this particular feature. The semantics of sharing node groups across projects was derived from how reservations are shared (take a look at the **shareSettings** field of the **Reservation** resource). Is there anything we can do with that at this point?

Also, I've stumbled upon some custom code that seems to be responsible for **shareSettings** modifications when updating the **Reservation** resource:
mmv1/templates/terraform/update_encoder/reservation.go.erb
Could you let me know if that is something we need for the **NodeGroup** resource as well?

Reply:
We don't need to add update_encoder for `shareSettings` in the resource `NodeGroup`. The resource NodeGroup has property `input: true`. That means that a new resource is created no matter what field has changed (from the code, node_template is an exception). 

https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/products/compute/api.yaml#L8794

Waiting for @c2thorn 's opinions.

Reply:
@idyczko thank you for pointing out the `shareSettings` field in Reservation. Considering that we've already gone forward with it there, we may as well keep consistent here.

@zli82016 is correct that a custom update encoder should not be needed since the resource cannot be updated in place at all. If this is no longer true, we can modify this in a separate PR.